dlt/common/data_writers/buffered.py

```diff
@@ -226,11 +227,20 @@ def _buffer_items_with_row_count(self, item: TDataItems) -> int:
                 new_rows_count = 1
         return new_rows_count
 
-    def _rotate_file(self, allow_empty_file: bool = False) -> DataWriterMetrics:
+    def _rotate_file(
```

Comment:
I would add a parameter `mark_compressed` instead of `is_imported_file`. if set to None, take the should_compress value, if set to true of false, respect that setting instead to decide wether to add this extension or not.